teckers/teckers/SessionController.swift

```diff
@@ -0,0 +1,281 @@
+                flag += 1
+                day.backgroundColor = UIColor.purple
+            }
+            else {
+                day.backgroundColor = UIColor.clear
+                day.labelDay.textColor = UIColor.black
+            }
+        }
+        else {
+            day.backgroundColor = UIColor.clear
+            day.labelDay.text = ""
+        }
+        day.layer.borderColor = UIColor.clear.cgColor
+        
+        return day
+    }
+    
+    func collectionView(_ collectionView: UICollectionView, didSelectItemAt indexPath: IndexPath) {
+        if (collectionView == Sessions) {
+            let dateFormatter = DateFormatter()
+            dateFormatter.dateFormat = "yyyy-MM-dd"
+            let dateSession = dateFormatter.date(from: sessions[indexPath.row].date)
+            let dateCurrent = dateFormatter.date(from: "\(yearCurrent)-\(monthCurrent)-\(dayCurrent)")
+            
+            if (dateSession?.description == dateCurrent?.description && sessions.count > 0) {
+                let alertNotification = Alert(title: "Asistencia", massage: "¿Asistirá a esta sesión?", type: 1)
+                present(alertNotification.show(), animated: true, completion: nil)
+            }
+        }
+        else if (collectionView == Calendar) {
+            let day = Calendar.cellForItem(at: indexPath)
+
+            if ((indexPath.row + 1) > (startOfMonth - 1)) {
+                if ((indexPath.row + 1  - startOfMonth + 1) == NSCalendar.current.component(.day, from: Date()) && monthCurrent == NSCalendar.current.component(.month, from: Date()) && yearCurrent == NSCalendar.current.component(.year, from: Date())) {
+                    day?.layer.borderColor = UIColor.black.cgColor
+                }
+                else {
+                    day?.layer.borderColor = UIColor.purple.cgColor
+                }
+                day?.layer.borderWidth = 2
+                day?.isSelected = true
+                dayCurrent = indexPath.row + 1 - startOfMonth + 1
+                sessionsDate = []
+                for d in sessions {
+                    let dateCurrent = "\(yearCurrent)-\(String(monthCurrent).left(total: 2, cadena: "0"))-\(String(dayCurrent).left(total: 2, cadena: "0"))"
+                    if d.date == dateCurrent {
+                        sessionsDate.append(d)
+                    }
+                }
+                sessionsValue = sessionsDate.count
+                Sessions.reloadData()
+            }
+        }
+    }
+    
+    func collectionView(_ collectionView: UICollectionView, didDeselectItemAt indexPath: IndexPath) {
+        if (collectionView == Sessions) {
+            let day = Sessions.cellForItem(at: indexPath)
+            day?.layer.borderColor = UIColor.clear.cgColor
+            day?.isSelected = false
+        }
+        let day = Calendar.cellForItem(at: indexPath)
+        day?.layer.borderColor = UIColor.clear.cgColor
+        day?.isSelected = false
+    }
+}
+
+extension String {
```

Comment:
this extension should be placed here .... String has nothing todo with Session management